modules/appgw/outputs.tf

```diff
@@ -2,3 +2,8 @@ output "public_ip" {
   description = "A public IP assigned to the Application Gateway."
   value       = azurerm_public_ip.this.ip_address
 }
+
+output "public_domain_name" {
+  description = "Public domain name assigned to the Application Gateway."
+  value       = azurerm_public_ip.this.fqdn
```

Comment:
in this form, when `var.domain_name_label` is not specified, you get the following input:
```
fqdn = {
  "public_appgw" = tostring(null)
}
```

Which works, but doesn't look nice ;) 

And we never know how this will be treated in future provider versions. So maybe wrap it in some condition like:

```
var.domain_name_label == null ? null : azurerm_public_ip.this.fqdn
```

Reply:
normal behaviour